pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1378,21 +1378,22 @@ func doInactivate(ctx *domainContext, status *types.DomainStatus, impatient bool
 			// This often fails with "X is an invalid domain identifier (rc=-6)"
 			// due to the DomainShutdown above, in which case
 			// the domain is already on the way down.
-			// In that case we wait
+			// In case of errors we should defer waitForDomainGone
+			// to delete task
```

Comment:
I didn't quite understand what "defer" means here, nor "delete task" since there is no task called "delete".'

To me it would be more clear if you said
In case of errors we proceeed directly to deleting the task, and after that we waitForDomainGone

Reply:
Done